src/main/java/org/opensearch/knn/index/query/filtered/KNNFloatQueryVector.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.filtered;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializer;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Implementation of KNNQueryVector with float data type
+ */
+public class KNNFloatQueryVector implements KNNQueryVector {
+    private final float[] queryVector;
+
+    public KNNFloatQueryVector(final float[] queryVector) {
+        this.queryVector = queryVector;
+    }
+
+    // Calculate similarity between queryVector and values in current position using given space type
+    public float score(final BinaryDocValues values, final SpaceType spaceType) throws IOException {
```

Comment:
nit: shall we add non null at required places?

Reply:
added